modules/WebPublishPlugin/src/main/java/net/clementlevallois/web/publish/plugin/controller/PublishingActions.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * author: Clément Levallois
+ */
+package net.clementlevallois.web.publish.plugin.controller;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.ResourceBundle;
+import static net.clementlevallois.web.publish.plugin.controller.GlobalConfigParams.*;
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.GraphController;
+import org.gephi.graph.api.GraphModel;
+import org.gephi.io.exporter.api.ExportController;
+import org.gephi.io.exporter.plugin.ExporterGEXF;
+import org.gephi.io.exporter.spi.CharacterExporter;
+import org.gephi.io.exporter.spi.Exporter;
+import org.gephi.project.api.ProjectController;
+import org.gephi.project.api.Workspace;
+import org.openide.util.Exceptions;
+import org.openide.util.Lookup;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author LEVALLOIS
+ */
+public class PublishingActions {
+
+    private static final ResourceBundle bundle = NbBundle.getBundle(GephiPluginDesktopLogic.class);
+
+    private static final int BYTES_IN_A_MEGABYTE = 1_048_576;
+
+    public static JsonObject getGexfAsString() {
```

Comment:
I don't get why you're using a `JsonObject` to pass data between this class and the client. Seems like your goal was to be able to pass the gexf data plus some kind of error codes with it. 
- The GEXF string seems to be converted back and forth multi times between JSON and String as a result
- Instead of error codes, a more standard way in Java would be to create your own exception